client/oidc.go

```diff
@@ -0,0 +1,56 @@
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+const (
+	oidcResourceEndpoint = "/admin/api/services/%d/proxy/oidc_configuration.json"
+)
+
+// OidcConfiguration fetches 3scale product oidc configuration
+func (c *ThreeScaleClient) OidcConfiguration(productID int64) (*OIDCConfiguration, error) {
```

Comment:
being an acronym probably a good idea to call it `OIDCConfiguration` (same with the other methods/places where the acronym appears)